### PR TITLE
#23 - Adds D-day based claim method

### PR DESCRIPTION
**Pull request by devpavan04**
### Description
This PR adds D-day based claim method along with a method to update the D-day of a safe.

### CODE REVIEW

File: contracts/components/Safes.sol [hunk 1]
```diff
@@ -112,13 +134,15 @@ contract Safes {
      * @param _safeId Id of the safe
      * @param _claimType Type of the claim
      * @param _signalingPeriod Signaling time window
+     * @param _DDay The timestamp after which the beneficiary can directly claim the safe
```
Comment (koshikraj) [inline contracts/components/Safes.sol:137]: Mention the unit of timestamp seconds/ ms

File: contracts/components/Safes.sol [hunk 2]
```diff
@@ -135,6 +159,7 @@ contract Safes {
             signalingPeriod: _signalingPeriod,
             endSignalTime: 0,
             latestSignalTime: 0,
+            dDay: block.timestamp + _DDay,
```
Comment (koshikraj) [inline contracts/components/Safes.sol:162]: Unlike `signalBased`, the DDay timestamp is an absolute time. _DDay itself should be a future timestamp.

File: contracts/components/Safes.sol [hunk 3]
```diff
@@ -180,6 +207,7 @@ contract Safes {
...
```
Comment (koshikraj) [inline contracts/components/Safes.sol:210]: Unlike signalBased, the DDay timestamp is an absolute time. _DDay itself should be a future timestamp.

File: contracts/components/Safes.sol [hunk 4]
```diff
@@ -257,4 +285,22 @@ contract Safes {
...
+     * @param _DDay The timestamp after which the beneficiary can directly claim the safe
+     */
+    function _updateDDay(string memory _safeId, uint256 _DDay)
+        internal
+        DDayUpdate(_safeId)
+        returns (bool)
+    {
+        Types.Safe memory safe = safes[_safeId];
+
+        safe.dDay = block.timestamp + _DDay;
```
Comment (koshikraj) [inline contracts/components/Safes.sol:301]: Unlike signalBased, the DDay timestamp is an absolute time. _DDay itself should be a future timestamp.

Comment (koshikraj) [inline contracts/components/Safes.sol:301]: Allow updation only if the DDay hasn't passed.

File: src/contracts/SafientMain.ts [hunk 5]
```diff
@@ -102,8 +102,8 @@ export class SafientMain {
    * @param creatorAddress Address of the creator who created the safe offchain
    * @param safeId Id of the safe
    * @param claimType Type of claim the inheritor has go through
-   * @param signalingPeriod Number of days within which the safe creator is willing to send a signal
-   * @param dDay The timestamp after which the beneficiary can directly claim the safe
+   * @param signalingPeriod The time window in milliseconds within which the creator wants to signal the safe in response to a claim on the safe
```
Comment (koshikraj) [inline src/contracts/SafientMain.ts:105]: Isn't the `signalingPeriod` denoted in in seconds?

Comment (devpavan04) [reply src/contracts/SafientMain.ts:105]: Yes

Comment (koshikraj) [reply src/contracts/SafientMain.ts:105]: Please update these @param docs then.

**Review by koshikraj — APPROVED** [review]

**Pull request merged**